src/qtt/measurements/scans.py

```diff
@@ -870,22 +870,35 @@ def _convert_scanjob_vec(self, station, steplength=None, sweeplength=None, stepv
         """
 
         def _update_step_size_from_length(data, length):
-            """ From desired number of data points determine the required step size """
+            """ From desired number of data points determine the required step size.
+            length > 1: to get 'inclusive' data with length = 1, start and end values must be the same and step value
+            must be 0 which are invalid values.
+            """
             if length is not None:
                 if length == 1:
-                    data['step'] = 0
+                    raise ValueError('length must be > 1')
                 else:
                     if 'range' in sweepdata:
                         data['step'] = data['range'] / (length-1)
                     else:
                         data['step'] = (data['end'] - data['start']) / (length-1)
 
-        def _calculate_sweepvalues(param, sweep_data) -> SweepFixedValues:
-            """ From parameter and sweep data create a SweepFixedValues object """
-            sweep_values = SweepFixedValues(param,
-                                            start=sweep_data['start'],
-                                            stop=sweep_data['end'],
-                                            step=sweep_data['step'])
+        def _calculate_sweepvalues(param, sweep_data, inclusive_end=False) -> SweepFixedValues:
+            """ From parameter and sweep data create a SweepFixedValues object. """
+            if sweep_data['step'] == 0.:
+                raise ValueError('step value may not be 0')
+
+            if inclusive_end:
+                epsilon = 1e-8
+                sweep_values = param[sweep_data['start']:(sweep_data['end'] + epsilon):sweep_data['step']]
+            else:
+                diff = abs(sweep_data['end'] - sweep_data['start'])
+                tolerance = 1e-10
+                if int(np.ceil(diff - tolerance)) == 0:
+                    raise ValueError('start and end values must differ at least 1 step value')
```

Comment:
Here you check that start and end are not equal (upto the tolerance), but the error message is different. Do you want to check `diff > sweep_data['step']` or do you want to update the error message?

Reply:
I changed the message because end-start doesn't need to be > step. I added a test for this situation. The text was from a previous situation with other circumstances. Good notice!